test/e2e/operator.go

```diff
@@ -321,3 +325,91 @@ var _ = XDescribe("ARO Operator - MachineSet Controller", func() {
 		}
 	})
 })
+
+var _ = Describe("ARO Operator - Azure Subnet Reconciler", func() {
+	var vnetName, location, resourceGroup string
+	var subnetsToReconcile []string
+	var dummynsg mgmtnetwork.SecurityGroup
+	ctx := context.Background()
+
+	// TODO (robryan) rm this func once default to on https://github.com/Azure/ARO-RP/issues/1735
+	enableReconcileSubnet := func() {
+		instance, err := clients.AROClusters.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		if !instance.Spec.Features.ReconcileSubnets {
+			instance.Spec.Features.ReconcileSubnets = true
+			_, err = clients.AROClusters.AroV1alpha1().Clusters().Update(ctx, instance, metav1.UpdateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		}
+	}
+	// Gathers vnet name, resource group, location, and adds master/worker subnets to list to reconcile.
+	gatherNetworkInfo := func() {
+		oc, err := clients.OpenshiftClustersv20200430.Get(ctx, vnetResourceGroup, clusterName)
+		Expect(err).NotTo(HaveOccurred())
+		location = *oc.Location
+
+		vnet, masterSubnet, err := subnet.Split(*oc.OpenShiftClusterProperties.MasterProfile.SubnetID)
+		Expect(err).NotTo(HaveOccurred())
+		_, workerSubnet, err := subnet.Split((*(*oc.OpenShiftClusterProperties.WorkerProfiles)[0].SubnetID))
+		Expect(err).NotTo(HaveOccurred())
+
+		subnetsToReconcile = append(subnetsToReconcile, masterSubnet, workerSubnet)
+
+		r, err := azure.ParseResourceID(vnet)
+		Expect(err).NotTo(HaveOccurred())
+		resourceGroup = r.ResourceGroup
+		vnetName = r.ResourceName
+	}
+
+	// Creates an empty NSG that gets assigned to master/worker subnets.
+	createDummyNSG := func() {
+		dummynsg = mgmtnetwork.SecurityGroup{
+			Location:                      to.StringPtr(location),
+			Name:                          to.StringPtr("dummynsg"),
+			Type:                          to.StringPtr("Microsoft.Network/networkSecurityGroups"),
+			SecurityGroupPropertiesFormat: &mgmtnetwork.SecurityGroupPropertiesFormat{},
+		}
+		err := clients.NetworkSecurityGroups.CreateOrUpdateAndWait(ctx, resourceGroup, "dummynsg", dummynsg)
+		Expect(err).NotTo(HaveOccurred())
+		dummynsg, err = clients.NetworkSecurityGroups.Get(ctx, resourceGroup, "dummynsg", "")
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	BeforeEach(func() {
+		enableReconcileSubnet()
+		gatherNetworkInfo()
+		createDummyNSG()
+	})
+	AfterEach(func() {
+		err := clients.NetworkSecurityGroups.DeleteAndWait(context.Background(), resourceGroup, "dummynsg")
+		if err != nil {
+			log.Warn(err)
+		}
+	})
+	Specify("ARO Operator - Azure Subnet Reconciler must reconcile list of subnets when NSG is changed", func() {
```

Comment:
No need to duplicate "ARO Operator - Azure Subnet Reconciler" - it will be printed from the `Describe` block.

```suggestion
	It("must reconcile list of subnets when NSG is changed", func() {
```